File: VeraPelle_front_standalone/src/app/core/models/product-dto.ts
// src/app/core/models/product-dto.ts

import {ProductVariantDTO} from '@app/core/models/product-variant-dto';

export interface ProductDTO {
  id?: number;
  name: string;
  description: string;
  price: number;
  stockQuantity: number;
  categoryIds: number[];    // array di ID di categorie
  colorIds: number[];       // ← array di ID di colori
  variants?: { colorId: number; stockQuantity: number }[];
}

File: VeraPelle/src/main/java/psw/verapelle/DTO/ProductDTO.java
package psw.verapelle.DTO;

import lombok.*;
import java.math.BigDecimal;
import java.util.List;

@Getter @Setter
@NoArgsConstructor @AllArgsConstructor
public class ProductDTO {
    private Long id;
    private String name;
    private String description;
    private BigDecimal price;
    /** @deprecated gestito tramite varianti colore */
    @Deprecated
    private int stockQuantity;
    private List<Long> categoryIds;

    /** Liste delle varianti colore con stock dedicato */
    private List<VariantDTO> variants;
}
